### PR TITLE
[12.0][MIG] partner fax

### PR DESCRIPTION
**Pull request by aitorbouzas**
What do we have here? Have I just migrated the most complex addon in OCA ever?
Yup.

### CODE REVIEW

File: partner_fax/__init__.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# © 2018 Apruzzese Francesco <f.apruzzese@apuliasoftware.it>
```
Comment (tafaRU) [inline partner_fax/__init__.py]: Please take the opportunity to replace the copyright simbol by `Copyright` according to https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L2

File: partner_fax/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# © 2018 Apruzzese Francesco <f.apruzzese@apuliasoftware.it>
```
Comment (tafaRU) [inline partner_fax/__manifest__.py]: same as above

File: partner_fax/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+{
+    'name': 'Partner fax',
+    'category': 'Extra Tools',
+    'summary': 'Add fax number on partner',
+    'version': '12.0.1.0.0',
+    'license': 'AGPL-3',
+    'author':  'Odoo Community Association (OCA)',
+    'website': 'https://odoo-community.org/',
```
Comment (tafaRU) [inline partner_fax/__manifest__.py]: please replace it by one of these: https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11

File: partner_fax/models/__init__.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# © 2018 Apruzzese Francesco <f.apruzzese@apuliasoftware.it>
```
Comment (tafaRU) [inline partner_fax/models/__init__.py]: Please avoid Copyright from `__init__` file according to https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

File: partner_fax/models/res_partner.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# © 2018 Apruzzese Francesco <f.apruzzese@apuliasoftware.it>
```
Comment (tafaRU) [inline partner_fax/models/res_partner.py]: same as above

File: partner_fax/views/res_partner.xml [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (tafaRU) [inline partner_fax/views/res_partner.xml]: Please remove `encoding="utf-8"` as it's already included in python 3!

**Review by tafaRU — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Some nitpicking, otherwise LGTM (also tested on runbot).
Don't forget to fixup the requested changes with the migration commit. BTW, this latter should be as `[MIG] partner_fax: Migration to 12.0` (see https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-12.0#technical-method-to-migrate-a-module-from-110-to-120-branch as reference).

File: partner_fax/views/res_partner.xml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+<odoo>
+
+    <record id="view_partner_simple_form_fax" model="ir.ui.view">
+        <field name="name">Add fax on partner</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_simple_form"/>
+        <field name="arch" type="xml">
+            <data>
+                <xpath expr="//field[@name='function']" position="after">
```
Comment (nikul-serpentcs) [inline partner_fax/views/res_partner.xml]: @aitorbouzas Here direct use `field`.

Ex. `<field name="function" position="after">`

File: partner_fax/views/res_partner.xml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        </field>
+    </record>
+
+    <record id="view_partner_short_form_fax" model="ir.ui.view">
+        <field name="name">Add fax on partner</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_short_form"/>
+        <field name="arch" type="xml">
+            <data>
+                <xpath expr="//field[@name='function']" position="after">
```
Comment (nikul-serpentcs) [inline partner_fax/views/res_partner.xml]: same here and Remove [data](https://github.com/OCA/partner-contact/pull/637/files#diff-3bcdbda29f030aded403bbdf43e36986R22)

File: partner_fax/views/res_partner.xml [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        </field>
+    </record>
+
+    <record id="view_partner_form_fax" model="ir.ui.view">
+        <field name="name">Add fax on partner</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="arch" type="xml">
+            <data>
+                <xpath expr="//field[@name='function']" position="after">
```
Comment (nikul-serpentcs) [inline partner_fax/views/res_partner.xml]: same here

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: partner_fax/__init__.py (hunk 1 above)
Comment (aitorbouzas) [reply partner_fax/__init__.py]: I dropped the line from this file since this is an __init__.py

File: partner_fax/__manifest__.py (hunk 2 above)
Comment (aitorbouzas) [reply partner_fax/__manifest__.py]: Done

File: partner_fax/__manifest__.py (hunk 3 above)
Comment (aitorbouzas) [reply partner_fax/__manifest__.py]: Done

File: partner_fax/models/__init__.py (hunk 4 above)
Comment (aitorbouzas) [reply partner_fax/models/__init__.py]: Dropped it, thanks.

**Comment by aitorbouzas** [comment]
@tafaRU made all the requested changes, thanks for the help!
@nikul-serpentcs thanks for pointing it out, it's resolved too.

File: partner_fax/__init__.py (hunk 1 above)
Comment (tafaRU) [reply partner_fax/__init__.py]: right!

**Review by tafaRU — APPROVED** [review]
Approved once travis is green. Thanks!

**Comment by aitorbouzas** [comment]
Green!

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM

**Comment by tafaRU** [comment]
@aitorbouzas, great!
FYI, before we could merge it: https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#please-respect-a-few-basic-rules

**Review by astirpe — APPROVED** [review]

**Comment by nikul-serpentcs** [comment]
@tafaRU @pedrobaeza If Done then Move forward!!

**Pull request merged**